airbyte-integrations/connectors/source-mixpanel/source_mixpanel/manifest.yaml

```diff
@@ -129,12 +129,16 @@ definitions:
       record_selector:
         $ref: "#/definitions/selector_empty_dpath"
         record_filter:
-          condition: "{{ record['created'] >= stream_state.created if stream_state.created else true }}"
+          condition: |
+            {%- set filter_date = stream_state.created or stream_slice.start_time -%}
+            "{{ record['created'] >= filter_date if filter_date else true }}"
```

Comment:
when would `filter_date` not be true?

I'd expect us to be able to simplify the filter to
`{{ record['created'] >= stream_interval["start_time"] }}`

Mid-term, we should stop exposing the state to connector developer because it's often misused and leads to bugs that are often difficult to troubleshoot. [Issue for reference](https://github.com/airbytehq/airbyte-internal-issues/issues/2209)

Reply:
I have rewritten this part with stream_interval usage